compiler/rustc_lint/src/traits.rs

```diff
@@ -114,10 +114,7 @@ impl<'tcx> LateLintPass<'tcx> for DropTraitConstraints {
         let hir::TyKind::TraitObject(bounds, _lifetime, _syntax) = &ty.kind else { return };
         for bound in &bounds[..] {
             let def_id = bound.trait_ref.trait_def_id();
-            if def_id.is_some_and(|def_id| cx.tcx.is_lang_item(def_id, LangItem::Drop))
-                // FIXME: ?Drop is not a thing.
-                && bound.modifiers != hir::TraitBoundModifier::Maybe
```

Comment:
:+1:

Technically speaking this is a behavior change I think, but one that doesn't matter imo:

```rs
// fail → pass (I think?)

#![feature(more_maybe_bounds)] // internal
#![deny(dyn_drop)]

fn main() {
    let _: dyn std::fmt::Debug + ?Drop;
}
```

Reply:
Yeah, we don't support `?Drop` so I don't know why we're special-casing it tbh.